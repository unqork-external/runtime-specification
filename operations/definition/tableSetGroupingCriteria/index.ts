export * from './tableSetGroupingCriteria.model'
export * from './tableSetGroupingCriteria.options'
