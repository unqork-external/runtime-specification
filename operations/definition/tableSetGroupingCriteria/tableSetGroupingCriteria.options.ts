import { Description, Required } from '@tsed/schema'

import { OneOrMany } from '../../../../decorators/schema/oneOrMany.decorator'

export class TableSetGroupingCriteriaOperationOptions {
  @Required()
  @Description('Key of the Table component to perform column grouping.')
  targetKey: string

  @Required()
  @Description('The label of the column(s) to group the Table component by.')
  @OneOrMany('string')
  groupBy: string | string[]
}
