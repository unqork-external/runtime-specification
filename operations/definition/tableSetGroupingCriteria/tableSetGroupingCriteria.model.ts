import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { TableSetGroupingCriteriaOperationOptions } from './tableSetGroupingCriteria.options'
import { Stable } from '../../../../decorators/stability/stable.decorator'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'
@Description('Sets columns to group on the Table component.')
@Stable()
@DiscriminatorValue(OperationTypes.TABLE_SET_GROUPING_CRITERIA)
export class TableSetGroupingCriteriaOperation extends Operation<OperationTypes.TABLE_SET_GROUPING_CRITERIA> {
  public type = OperationTypes.TABLE_SET_GROUPING_CRITERIA as const

  @Required()
  public options: TableSetGroupingCriteriaOperationOptions = new TableSetGroupingCriteriaOperationOptions()
}
