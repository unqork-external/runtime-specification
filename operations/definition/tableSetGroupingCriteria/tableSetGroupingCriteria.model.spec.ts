import { TableSetGroupingCriteriaOperation } from './tableSetGroupingCriteria.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableSetGroupingCriteria Tests', () => {
  let validate, schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(TableSetGroupingCriteriaOperation)
    validate = response.validate
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
