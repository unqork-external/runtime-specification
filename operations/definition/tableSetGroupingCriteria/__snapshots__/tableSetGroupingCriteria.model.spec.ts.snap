// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableSetGroupingCriteria Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "TableSetGroupingCriteriaOperationOptions": {
      "properties": {
        "groupBy": {
          "anyOf": [
            {
              "type": "string",
            },
            {
              "items": {
                "type": "string",
              },
              "type": "array",
            },
          ],
          "description": "The label of the column(s) to group the Table component by.",
        },
        "targetKey": {
          "description": "Key of the Table component to perform column grouping.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "groupBy",
      ],
      "type": "object",
    },
  },
  "description": "Sets columns to group on the Table component.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/TableSetGroupingCriteriaOperationOptions",
    },
    "type": {
      "const": "TABLE_SET_GROUPING_CRITERIA",
      "examples": [
        "TABLE_SET_GROUPING_CRITERIA",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
